### DISCUSSION TITLE
Plasma THCAD-xxx support

### DISCUSSION THREAD

**Discussion by electroklas** [grblHAL/core#640] [Q&A] [answered]
Is there a way to utilse a Mesa THCAD-xxx arc voltage monitor with grblHAL ?

The THCAD is built arround a voltage to frequency converter instead of a a/d... This means it only requires a digital in port and no analog input, but of course its all up to sw instead...

(Upgrading my PlasmaCNC from grbl to grblHAL with a Teensy 4.1, (T41U5XBB) and had a THCAD laying arround...)

**Comment by terjeio** [comment]
> but of course its all up to sw instead...

Yes, it is a matter of converting the frequency to a value representing the voltage and making it available via the [ioports_analog](https://github.com/grblHAL/iMXRT1062/blob/master/grblHAL_Teensy4/src/ioports_analog.c) interface.

**Comment by electroklas** [comment]
Ok, tecnically possible, but not implemented to this date…is my take from that. :)

I2C A/D card ordered from Brookwood design..

**Reply by terjeio** [reply, marked as answer]
You can implement it?
The spindle sync counter can be used to count the pulses. A 32 bit counter is needed and the iMXRT1062 counter used for spindle sync is 32 bit... The systick interrupt handler (1ms interval) can be extended to provide the gating (sampling) of the counter.

**Reply by electroklas** [reply]
:) ooh... My sw skills are not sufficient for that task unfortionately.... Im more of a sustem/hw person...

**Comment by terjeio** [comment]
FYI tentative [THCAD2 support](https://github.com/grblHAL/RP2040/blob/master/thcad2.c) has been added to the RP2040 driver, I may add it to some of the other drivers later.
